vcd/datasource_vcd_vapp.go

```diff
@@ -78,6 +78,30 @@ func datasourceVcdVApp() *schema.Resource {
 					},
 				},
 			},
+			"inherited_metadata": {
```

Comment:
I suggest making a variable of this structure, using
`origin_id`, `origin_name`, and `origin_type` instead of the current strings.
This way, it could be used by both the vApp and the VM

Reply:
The abstract names would be too confusing. Right now it's very clear what these fields represent, comparison:
- `vapp_origin_name` ("Name of the **vApp** from which this vApp was generated")
- `vm_origin_name` ("Name of the **VM** from which this VM was generated")

Without these prefixes:
- `origin_name` - it's very vague and user has to know a lot up front.

So, let's leave the names as is.

Reply:
I've changed the block into a `TypeMap`, so the metadata is added as it comes from VCD, instead of transforming it.